homeassistant/components/sensor/swiss_hydrological_data.py

```diff
@@ -151,38 +140,29 @@ class HydrologicalData:
     def __init__(self, station):
         """Initialize the data object."""
         self.station = station
-        self.measurings = None
+        self.measurements = None
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data from hydrodata.ch."""
-        import xmltodict
-
-        details = {}
-        try:
-            response = requests.get(_RESOURCE, timeout=5)
-        except requests.exceptions.ConnectionError:
-            _LOGGER.error("Unable to retrieve data from %s", _RESOURCE)
-
-        try:
-            stations = xmltodict.parse(response.text)['AKT_Data']['MesPar']
-            # Water level: Typ="02", temperature: Typ="03", discharge: Typ="10"
-            for station in stations:
-                if str(self.station) != station.get('@StrNr'):
-                    continue
-                for data in ['02', '03', '10']:
-                    if data != station.get('@Typ'):
-                        continue
-                    values = station.get('Wert')
-                    if values is not None:
-                        details[data] = {
-                            'current': values[0],
-                            'max': list(values[4].items())[1][1],
-                            'mean': list(values[3].items())[1][1]}
-
-                    details['location'] = station.get('Name')
-                    details['update_time'] = station.get('Zeit')
-
-            self.measurings = details
-        except AttributeError:
-            self.measurings = None
+        from swisshydrodata import SwissHydroData
+
+        data = {}
+
+        shd = SwissHydroData()
+        shd.load_station_data(self.station)
+
+        data["temperature"] = shd.get_latest_temperature()
+        data["level"] = shd.get_latest_level()
+        data["discharge"] = shd.get_latest_discharge()
+        data["min_temperature"] = shd.get_min_temperature()
+        data["min_level"] = shd.get_min_level()
+        data["min_discharge"] = shd.get_min_discharge()
+        data["max_temperature"] = shd.get_max_temperature()
+        data["max_level"] = shd.get_max_level()
+        data["max_discharge"] = shd.get_max_discharge()
+        data["mean_temperature"] = shd.get_mean_temperature()
+        data["mean_level"] = shd.get_mean_level()
+        data["mean_discharge"] = shd.get_mean_discharge()
```

Comment:
I'm not so sure that this is the right approach resource-wise. There will be a dozen requests for every update to get the data instead of one that is sharing the data.

You want to change the sensor style (single sensors instead of attributes), thus you should only get the data, e. g., for `mean_temperature` that's needed and not everything that is available. BTW, there is a chance (https://github.com/home-assistant/architecture/issues/100) that it needs to be updated/changed again soon.